src/test/scala/io/iohk/ethereum/jsonrpc/CheckpointingServiceSpec.scala

```diff
@@ -0,0 +1,68 @@
+package io.iohk.ethereum.jsonrpc
+
+import akka.actor.ActorSystem
+import akka.testkit.{TestKit, TestProbe}
+import io.iohk.ethereum.blockchain.sync.regular.RegularSync.NewCheckpoint
+import io.iohk.ethereum.domain.{Block, BlockBody, BlockchainImpl}
+import io.iohk.ethereum.jsonrpc.CheckpointingService._
+import io.iohk.ethereum.{Fixtures, NormalPatience}
+import org.scalacheck.Gen
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+
+class CheckpointingServiceSpec
+  extends TestKit(ActorSystem("CheckpointingServiceSpec_System"))
+    with AnyFlatSpecLike
+    with MockFactory
+    with ScalaFutures
+    with NormalPatience
+    with ScalaCheckPropertyChecks
+    with Matchers {
+
+  "CheckpointService" should "get latest block (at a correct checkpointing interval) from Blockchain" in new TestSetup {
```

Comment:
Should we maybe add a test for the case a reorg happened?

Reply:
Should be doable by setting expectations on blockchain accordingly, right?